From 8b029825f2b3d7767ddd87c931472171f137a976 Mon Sep 17 00:00:00 2001
From: "A.I" <ailis@paw.zone>
Date: Sun, 30 Dec 2018 18:48:50 -0800
Subject: [PATCH] More buffer to read/write faster

128K = VM_MAX_READAHEAD * 1024
---
 regops.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/regops.c b/regops.c
index 3b42ead..4501a86 100644
--- a/regops.c
+++ b/regops.c
@@ -42,8 +42,8 @@ static void *alloc_bounce_buffer(size_t * tmp_sizep, PRTCCPHYS physp, size_t
 
 	/* try for big first. */
 	tmp_size = RT_ALIGN_Z(xfer_size, PAGE_SIZE);
-	if (tmp_size > 16U * _1K)
-		tmp_size = 16U * _1K;
+	if (tmp_size > 128U * _1K)
+		tmp_size = 128U * _1K;
 	tmp = kmalloc(tmp_size, GFP_KERNEL);
 	if (!tmp) {
 		/* fall back on a page sized buffer. */
-- 
2.17.2 (Apple Git-113)

